### ISSUE TITLE
Unapproved comments page redirects away after submission

### ISSUE THREAD

**Issue by alanmels** [backdrop/backdrop-issues#4314]
**Description of the bug**
After submission of the **Unapproved comments** page (`admin/content/comment/approval`) it redirects away to the **Published comments** page (`admin/content/comment`), which could cause inattentive users to have an impression they are still on the same page and to go ahead to delete legitimate comments. 

It is especially dangerous when users are dealing with massive spam comments and when they want to stay at the same page and keep mass-selecting and deleting such comments until they are all gone.

**Steps To Reproduce**
To reproduce the behavior:
1. Make sure the "Skip comment approval" on `admin/config/people/permissions` is unchecked for a user role, for example, for ANONYMOUS role. Logout.
2. Post some comments as an ANONYMOUS user, so your comments are not yet approved.
3. Login as admin and go to the **Unapproved comments** page (`admin/content/comment/approval`), mass select comments and delete them and it redirects away to another page instead of staying at the same place.

**Actual behavior**
![out2](https://user-images.githubusercontent.com/56604405/74801438-d080e600-5293-11ea-8f78-b978bb4b55a2.gif)

**Expected behavior**

The expected behavior is to stay at the same page unless/until you intentionally browse away.

**Comment by izmeez** [comment]
@alanmels Were you able to resolve this?

**Comment by izmeez** [comment]
I have added a PR that works on local testing but is failing on 3 automated tests that are related to the comment UI, maybe this is to be expected because the patch changes the behavior of the comment UI?

**Comment by argiepiano** [comment]
Thanks for tackling this! Two comments:

1. The easier solution to this in `comment_admin_overview_submit()` is to just remove the line `$form_state['redirect'] = 'admin/content/comment';`  instead of the `if()` statement. This will make the site stay on the current page after submitting (and in the unlikely case that there is a `?destination=` query, it will respect that).
2. As for the tests: it looks like the new behavior has introduced a regression, or more possibly, that the tests need to be rewritten to accommodate the lack of the forced redirect. You'll need to install simpletest locally and manually run that specific test to trace the problem

**Comment by izmeez** [comment]
@argiepiano Tested your suggestion and you are absolutely correct, less is more, just took the line out! I'll have to submit a new PR.

As for tests, that's another whole thing for me, never had time to fit that into my time available.

**Comment by izmeez** [comment]
The new PR is up. It was so simple I felt the need to add a comment. Feedback welcome. Tests are failing as before and as expected.

**Comment by argiepiano** [comment]
Yes, simple. I don't think the added comment in the docblock is needed. A much more important priority is getting the tests passing... the PR can't be merged until that's resolved.

**Comment by izmeez** [comment]
I decided to take another look at testing. I installed the [SimpleTester](https://github.com/backdrop-contrib/simpletester) module and taking a look, again considering the issue I posted before, https://github.com/backdrop-contrib/simpletester/issues/1. Now to learn what I can find.

**Comment by argiepiano** [comment]
Test passing now 🎉 

There is one little inconsistency now. When you approve, you remain in the Unpublished comments screen, but if you delete an unpublished message, you are redirected to the Published comments screen. I'd suggest that both should remain in the unpublished one

**Comment by izmeez** [comment]
The tests are passing now and I have reduced the comments.

> I'd suggest that both should remain in the unpublished one

Yes, that was the main reason. Are you saying this change doesn't fix that?

**Comment by argiepiano** [comment]
It only fixed the issue when you **approve** messages, but not when you **delete** them.

**Comment by izmeez** [comment]
I sees another line:
>   $form_state['redirect'] = 'admin/content/comment';

in the delete multiple. I'll remove that and see what tests break.

**Comment by izmeez** [comment]
That is done and the tests pass. There may be a desire to add more tests to affirm we are staying on the right url.

**Comment by argiepiano** [comment]
LGTM!

**Comment by quicksketch** [comment]
Thanks @izmeez, @argiepiano, and @alanmels! Merged https://github.com/backdrop/backdrop/pull/5287 into 1.x and 1.34.x.
